src/LayerPlanBuffer.cpp

```diff
@@ -17,16 +17,55 @@ void LayerPlanBuffer::flush()
     }
     while (!buffer.empty())
     {
-        buffer.front().writeGCode(gcode);
+        buffer.front()->writeGCode(gcode);
         if (CommandSocket::isInstantiated())
         {
             CommandSocket::getInstance()->flushGcode();
         }
         buffer.pop_front();
     }
-    
 }
 
+void LayerPlanBuffer::addConnectingTravelMove()
+{
+    auto newest_layer_it = --buffer.end();
+    auto prev_layer_it = newest_layer_it;
+    prev_layer_it--;
+    Point destination;
+    {
+        const LayerPlan* newest_layer = *newest_layer_it;
+        assert(newest_layer->extruder_plans.front().extruder == (*prev_layer_it)->extruder_plans.back().extruder);
+        const ExtruderPlan& extruder_plan_above = newest_layer->extruder_plans.front();
+        if (extruder_plan_above.paths.size() == 0)
+        {
+            logWarning("There are empty layers (or layers with empty extruder plans) in the print! Temperature control and cross layer travel moves might suffer.\n");
+            return;
+        }
+        const GCodePath& path_above = extruder_plan_above.paths[0];
+        assert(path_above.points.size() == 1 && "The first move planned should have been a boguous direct travel move");
+        Point first_location_new_layer = path_above.points[0];
+        destination = first_location_new_layer;
+    }
+    LayerPlan* prev_layer = *prev_layer_it;
+    prev_layer->addTravel(destination);
+}
+
+void LayerPlanBuffer::processFanSpeedLayerTime()
+{
+    auto newest_layer_it = --buffer.end();
+    Point starting_position(0, 0);
+    if (buffer.size() >= 2)
+    {
+        auto prev_layer_it = newest_layer_it;
+        prev_layer_it--;
+        const LayerPlan* prev_layer = *prev_layer_it;
+        starting_position = prev_layer->getLastPosition();
+    }
+    LayerPlan* newest_layer = *newest_layer_it;
```

Comment:
refactored function such that its not possible any more
(though the pointer in the buffer is still assumed to point to an actual layer)